#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.20141703764237692 0.26439209020080884 0.9431479203108157 1.3588911453274435
  position -0.5450040901881873 -2.6571251173007777 1.6098598007358147
  follow "robot"
  followType "Pan and Tilt Shot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 1
}
Robot {
  rotation 0 0 1 3.14159
  children [
    DEF SLIDER_CART SliderJoint {
      jointParameters JointParameters {
        axis 1 0 0
        dampingConstant 1
      }
      device [
        LinearMotor {
          name "horizontal_motor"
          maxForce 40
        }
        PositionSensor {
          name "horizontal position sensor"
        }
      ]
      endPoint DEF CART_SOLID Solid {
        translation 0 0 0.06
        children [
          DEF CART_SHAPE Shape {
            appearance PBRAppearance {
              baseColor 0.244343 0.704051 0.175692
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.25 0.1 0.1
            }
          }
          DEF HIP HingeJoint {
            jointParameters HingeJointParameters {
              position 0.000161402
              axis 0 1 0
              anchor 0 0 0.03
            }
            device [
              PositionSensor {
                name "hip"
              }
            ]
            endPoint DEF IP_SOLID Solid {
              translation 0 -0.061 0.33000000000000007
              rotation 0 1 0 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.849683 0.0707256 0.0939345
                    roughness 1
                    metalness 0
                  }
                  geometry DEF IP_BOX_SHAPE Box {
                    size 0.05 0.02 0.6
                  }
                }
              ]
              boundingObject USE IP_BOX_SHAPE
              physics Physics {
                density -1
                mass 0.23
                centerOfMass [
                  0 0.061 -0.27
                ]
              }
            }
          }
        ]
        contactMaterial "robot_basis"
        boundingObject USE CART_SHAPE
        physics Physics {
          density -1
          mass 1.2
        }
      }
    }
  ]
  boundingObject Box {
    size 2 0.1 0.01
  }
  physics Physics {
    density -1
    mass 1.2
    damping Damping {
      linear 1
    }
  }
  controller "pid_ip"
  supervisor TRUE
}
